coap/transport.py

```diff
@@ -0,0 +1,84 @@
+import asyncio
+import logging
+
+from aioquic.asyncio import connect, serve
+from aioquic.asyncio.protocol import QuicConnectionProtocol
+from aioquic.quic.configuration import QuicConfiguration
+from aioquic.quic.events import HandshakeCompleted, StreamDataReceived
+
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class ProxyClientProtocol(QuicConnectionProtocol):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._next_stream_id = 0  # Start with 0 for client-initiated streams
+        self.responses = asyncio.Queue()
+
+    def quic_event_received(self, event):
+        if isinstance(event, HandshakeCompleted):
+            logger.info("QUIC handshake completed")
+            self._stream_id = self._quic.get_next_available_stream_id()
+
+        elif isinstance(event, StreamDataReceived):
+            logger.info(f"Received: {event.data} on stream {event.stream_id}")
+            asyncio.ensure_future(self.responses.put(event.data))
+
+    async def send_request_over_quic(self, payload):
+        logger.info(f"Sending to server-proxy, payload: '{payload}'")
+        self._quic.send_stream_data(self._stream_id, payload)
+        self.transmit()
+
+        response = await self.responses.get()
+        return response
+
+
+class ProxyServerProtocol(QuicConnectionProtocol):
+    def __init__(self, *args, packet_handler=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._stream_id = None
+        self.packet_handler = packet_handler
+
+    def quic_event_received(self, event):
+        if isinstance(event, HandshakeCompleted):
+            self._stream_id = self._quic.get_next_available_stream_id()
+
+        elif isinstance(event, StreamDataReceived):
+            asyncio.ensure_future(self.handle_quic_request(event.data, event.stream_id))
+
+    async def handle_quic_request(self, data, stream_id):
+        logger.info(f"Received data over QUIC: {data}")
+
+        response = await self.packet_handler(data)
+
+        # Send the CoAP response back over QUIC
+        self._quic.send_stream_data(stream_id, response.encode())
+        self.transmit()
+
+
+class ProxyServerProtocolFactory:
```

Comment:
Factory class which helps pass packet handler function. This makes packet handling flexible. We can use this later for MQTT as well (hopefully).